sdk/core/azure-core/azure/core/credentials.py

```diff
@@ -3,21 +3,56 @@
 # Licensed under the MIT License. See LICENSE.txt in the project root for
 # license information.
 # -------------------------------------------------------------------------
-from typing import Any, NamedTuple, Optional
+from typing import Any, NamedTuple, Optional, TypedDict
 from typing_extensions import Protocol, runtime_checkable
 
 
 class AccessToken(NamedTuple):
     """Represents an OAuth access token."""
 
     token: str
+    """The token string."""
     expires_on: int
-    refresh_on: Optional[int] = None
+    """The token's expiration time in Unix time."""
 
 
-AccessToken.token.__doc__ = """The token string."""
-AccessToken.expires_on.__doc__ = """The token's expiration time in Unix time."""
-AccessToken.refresh_on.__doc__ = """When the token should be refreshed in Unix time."""
+class AccessTokenInfo:
+    """Information about an OAuth access token.
+
+    This class is an alternative to `AccessToken` which provides additional information about the token.
+
+    :param str token: The token string.
+    :param int expires_on: The token's expiration time in Unix time.
+    :keyword int refresh_on: Specifies the time, in Unix time, when the cached token should be proactively
+        refreshed. Optional.
+    """
+
+    token: str
+    """The token string."""
+    expires_on: int
+    """The token's expiration time in Unix time."""
+    refresh_on: Optional[int]
+    """Specifies the time, in Unix time, when the cached token should be proactively refreshed. Optional."""
+
+    def __init__(self, token: str, expires_on: int, *, refresh_on: Optional[int] = None) -> None:
+        self.token = token
+        self.expires_on = expires_on
+        self.refresh_on = refresh_on
+
+    def __repr__(self) -> str:
+        return f"AccessTokenInfo(token='{self.token}', expires_on={self.expires_on}, refresh_on={self.refresh_on})"
+
+
+class TokenRequestOptions(TypedDict, total=False):
```

Comment:
We don't need to put well-known options into the option bag.

Did I mis-understand?

Reply:
I think it makes sense to aggregate all optional token request options in a single container (similar to other languages: [example 1](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/core-auth/src/tokenCredential.ts#L27), [example 2](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/internal/exported/exported.go#L56)), keeping things more organized.

Reply:
Other languages prefer option bags, but Python prefers flattened arguments, right? By flattening them, we can provide better intellisense.

Reply:
We've established we need an options bag in this case to prevent frequently changing protocol method signatures. And also to prevent issues with implementors of the protocol passing in arbitrary kwargs to transports. The PoP changes appears like it will introduce 3-4 new params. 

If we are putting some optional params in this container class, we might as well put all of them in it. To me, the intellisense experience still works fine since you can still dig down to see what the parameters of the TypedDict are. The majority of SDK users wouldn't be calling this method directly anyway, since it's more of a policy thing.

Reply:
e.g. if I (user) want to add "transport" in the options, how can I achieve it?

Reply:
We don't support that since that is a Credential constructor level option. New options can be added as-needed if there is a strong enough requirement for them.

Reply:
Then how about "timeout"? It is supported today.

Reply:
Sure, this will still be supported in `get_token`. 

However, our policies don't propagate these transport level args anyway since we just do a `self._credential.get_token(*self._scopes)` in our `on_request` methods. Our policies are the only users of `get_token_info` , and if for some reason, a user really needs these transport configurations at the token request level, we can always add it later (and maybe in a cleaner way rather than just passing all kwargs). At this stage, I don't think we need it.

Reply:
Let's take one step back.

What's the user experience of pop token?

Reply:
Outlined in the design gist: https://gist.github.com/christothes/98e0a1c554f180c5ed0c2094d3b93c08#api-changes-overview . Services that support PoP will add the new PopPolicy to their SDK pipelines. I don't think end users of our SDKs will really even notice or have to configure anything for PoP auth. 

The new classes established in this PR can (and will) be extended to support all the new configuration in any case. From our side, it'll be adding the handful of new options to `TokenRequestOptions`, `token_type` to `AccessTokenInfo`, and eventually adding a new pop policy. But again, I want to keep Pop changes separate from this PR.

Reply:
I was asking for end user experience.

Do you mean user does not need to know if it is pop token or not?

We use pop token as long as it is available?

Reply:
IOW, the key question is if pop is on by default (as long as it is available).

Reply:
Right, from what I understand from the .NET implementation and the design doc, the PopAuthPolicy will only be added to services that support Pop auth. I believe, right now, only MS Graph supports this? So services that don't work with Pop likely wont even have the Pop policy in their pipelines.

And I believe there should still be a fallback to the BearerTokenCredentialPolicy if for whatever reason that fails. Will need to double check with Chris on that. But that's what I recall.